### ISSUE TITLE
(--check-unsat-cores, --check-proofs)  Fatal failure at src/proof/proof_node_updater.cpp:173

### ISSUE THREAD

**Issue by wintered** [cvc5/cvc5#9172]
[8635ad0](https://github.com/cvc5/cvc5/commit/8635ad0970938dec262234f9107ab70b2c074a32)
```
$cvc5 -q --strings-exp --check-proofs --check-unsat-cores delta.out.smt2
Fatal failure within void cvc5::internal::ProofNodeUpdater::processInternal(
std::shared_ptr<cvc5::internal::ProofNode>,  std::vector<cvc5::internal::NodeTemplate<true> >&)
 at /local/disk1/dominik- exp/bin/.solvers/cvc5/src/proof/proof_node_updater.cpp:173
Unhandled case encountered ProofNodeUpdater::processInternal: cyclic proof! 
(use --proof-check=eager)

[1]    55860 abort      cvc5 -q --strings-exp --check-proofs --check-unsat-cores delta.out.smt2
$cat delta.out.smt2 
(declare-fun b () String)
(assert (> (ite (str.prefixof "-" b) 1 0) 1))
(assert (ite (str.prefixof "-" b) false (ite (= 0 (str.to_int (str.at b 
            (+ (str.len b) (- 1))))) false true)))
(assert (= (str.len b) 1))
(check-sat)
```

**Comment by ajreynol** [comment]
Likely a duplicate of https://github.com/cvc5/cvc5/issues/9156.

**Comment by ajreynol** [comment]
Fixed by #9242.
